src/main/java/Duke.java

```diff
@@ -1,10 +1,115 @@
+    public static final String BOTTOM_LINE = "\n____________________________________________________________";
+    public static final String LIST_HEADER = FRONT_SPACING + "Here are the tasks in your list:\n";
+    public static final String INVALID_TASK = FRONT_SPACING + "Invalid task to insert";
+    public static final String TASK_HEADER = FRONT_SPACING + "Got it. I've added this task: \n";
+    public static final String TASK_FOOTER_FIRST_PART = FRONT_SPACING + "Now you have ";
+    public static final String TASK_FOOTER_SECOND_PART = " tasks in the list.";
+    public static final String INVALID_TASK_NUMBER = "Invalid task number";
+    public static final String MARKED_DONE_HEADER = "Nice! I've marked this task as done:\n";
+    public static final String GREETING_HEADER_TOP = FRONT_SPACING + "Hello! I'm Duke\n";
+    public static final String GREETING_HEADER_BOTTOM = FRONT_SPACING + "What can I do for you?";
+    public static final String BYE_HEADER = FRONT_SPACING + "Bye Bye! Hope to see you again soon!";
+
+
+    public static ArrayList<Task> allActions = new ArrayList<>();
+
+    /**
+     * Printing information within bounding lines
+     */
+    public static void printWithLines(String stringToPrint) {
+        String finalString = TOP_LINE + stringToPrint + BOTTOM_LINE;
+        System.out.println(finalString);
+    }
+
+    /**
+     * Listing all stored actions
+     */
+    public static void listAllActions() {
+        String fullList = LIST_HEADER;
+
+        for (int i = 0; i < allActions.size(); i++) {
+            fullList += String.format("%s%s.%s", FRONT_SPACING, (i + 1), allActions.get(i).getFullTask());
+
+            if (i != allActions.size() - 1) {
+                fullList += NEW_LINE;
+            }
+        }
+        printWithLines(fullList);
+    }
+
+    /**
+     * Storing new action into Array List
+     */
+    public static void addAction(String command, String userInput) {
+        int firstSpaceIndex = userInput.indexOf(" ");
+        String description = userInput.substring(firstSpaceIndex + 1);
+
+        Task newTask = new Task(userInput);
+
+        if (command.equals("todo")) {
+            newTask = new Todo(description);
+        } else if (command.equals("deadline")) {
+            String[] details = description.split(" /by ");
+            newTask = new Deadline(details);
+        } else if (command.equals("event")) {
+            String[] details = description.split(" /at ");
+            newTask = new Event(details);
+        }
+
+        allActions.add(newTask);
+        String taskConfirmation = newTask.getFullTask();
+
+        if (taskConfirmation.equals("")) {
+            printWithLines(INVALID_TASK);
+        } else {
+            printWithLines(TASK_HEADER
+                    + FRONT_SPACING + FRONT_SPACING + taskConfirmation + NEW_LINE
+                    + TASK_FOOTER_FIRST_PART + allActions.size() + TASK_FOOTER_SECOND_PART);
```

Comment:
I like this part that you consider another possible situation to make your code more completed.